doc/stdlib/linalg.rst

```diff
@@ -21,9 +21,9 @@ Linear algebra functions in Julia are largely implemented by calling functions f
 .. function:: \\(A, B)
    :noindex:
 
-   Matrix division using a polyalgorithm. For input matrices ``A`` and ``B``, the result ``X`` is such that ``A*X == B`` when ``A`` is square.  The solver that is used depends upon the structure of ``A``.  A direct solver is used for upper- or lower triangular ``A``.  For Hermitian ``A`` (equivalent to symmetric ``A`` for non-complex ``A``) the ``BunchKaufman`` factorization is used.  Otherwise an LU factorization is used. For rectangular ``A`` the result is the minimum-norm least squares solution computed by a pivoted QR factorization of ``A`` and a rank estimate of A based on the R factor.
+   Matrix division using a polyalgorithm. For input matrices ``A`` and ``B``, the result ``X`` is such that ``A*X == B`` when ``A`` is square.  The solver that is used depends upon the structure of ``A``.  A direct solver is used for upper or lower triangular ``A``.  For Hermitian ``A`` (equivalent to symmetric ``A`` for non-complex ``A``) the ``BunchKaufman`` factorization is used.  Otherwise an LU factorization is used. For rectangular ``A`` the result is the minimum-norm least squares solution computed by a pivoted QR factorization of ``A`` and a rank estimate of A based on the R factor.
 
-   When ``A`` is sparse, a similar polyalgorithm is used. For indefinite matrices, the LDLt factorization does not use pivoting during the numerical factorization and therefore the procedure can fail even for invertible matrices.
+   When ``A`` is sparse, the LU factorization is computed unless the matrix is triangular, in which case a direct solver is preferred. For indefinite matrices, the LDLt factorization does not use pivoting during the numerical factorization and therefore the procedure can fail even for invertible matrices.
```

Comment:
Indeed this is a bit confusing and I think that the method you are pointing to eventually should go away. It is not that method which is called when calling backslash on a sparse array. See
```jl
julia> @which \(sprandn(3,3,0.5), ones(3))
\{T}(A::AbstractArray{T,2}, B::Union{AbstractArray{T,1},AbstractArray{T,2}}) at linalg/generic.jl:311
```
which is defined here https://github.com/JuliaLang/julia/blob/50cf3537c548b15e9e708f31771ea7b78c101a45/base/linalg/generic.jl#L310-L315 and it calls `factorize`.